### PR TITLE
add RDS MariaDB e2e tests

### PR DESCRIPTION
**Pull request by GavinFrazar**
This PR adds RDS MariaDB discovery, connection, and auto user provisioning tests.

stacked on https://github.com/gravitational/teleport/pull/40065 and will be automatically rebased onto master when that merges.

We only added auto user provisioning in v14 for mariadb, but ~~I'm backporting to v13 for discovery and connection tests.~~ we only added e2e test in v14. Soon to be out of support anyway. I'll only backport to v14+.

### CODE REVIEW

File: e2e/aws/rds_test.go [hunk 1]
```diff
@@ -712,3 +794,44 @@ func waitForMySQLAutoUserDrop(t *testing.T, conn *mySQLConn, user string) {
...
+		if !assert.NoError(c, err) {
+			return
+		}
+		if !assert.Equal(c, 1, result.RowNumber(), "user %q should not be able to login after deactivating", user) {
+			result.Close()
+			return
+		}
+		result.Close()
+
+		result, err = conn.Execute("SELECT 1 FROM mysql.roles_mapping AS u WHERE u.user = ? AND u.role != 'teleport-auto-user'", user)
```
Comment (greedy52) [inline e2e/aws/rds_test.go:821]: nit: may also check ` Admin_option = 'N'`.

**Review by greedy52 — COMMENTED** [review]
Overall LGTM. Same comment as the other PR that we need to verify db_roles are assigned to the user.

**Comment by GavinFrazar** [comment]
@greedy52 let me know if there's any issues with doing https://github.com/gravitational/teleport/pull/40066/commits/dbe00fce1b790e1841812788c0217b186e3be2e7

File: lib/srv/db/postgres/sql/delete-user.sql [hunk 2]
```diff
@@ -9,7 +9,7 @@ BEGIN
         RAISE NOTICE 'User has active connections';
     ELSE
         BEGIN
-            EXECUTE FORMAT('DROP USER %I', username);
+            EXECUTE FORMAT('DROP USER IF EXISTS %I', username);
```
Comment (greedy52) [inline lib/srv/db/postgres/sql/delete-user.sql:12]: Thanks for the fix. Let's also apply this to `redshift-delete-user.sql`
https://docs.aws.amazon.com/redshift/latest/dg/r_DROP_USER.html

**Review by greedy52 — APPROVED** [review]

**Review by probakowski — APPROVED** [review]

**Comment by GavinFrazar** [comment]
I can't figure out why the gha runners are being blocked by the MariaDB instance.

performance_schema is disabled by default, so it wouldn't even be possible to look at that.

I figured out that a host only gets blocked for connection failure, *not* auth failure. This can be simulated by just running `nc -q 0 -v <host> <port>` - the protocol handshake is server initiated and since we start talking to the db and then leave, it counts as a connection failure.

Default `max_connect_errors` is 100. I'm going to bump that value up a lot and then just wait for the db to get recreated over the weekend.

**Comment by greedy52** [comment]
> I can't figure out why the gha runners are being blocked by the MariaDB instance.
> 
> performance_schema is disabled by default, so it wouldn't even be possible to look at that.
> 
> I figured out that a host only gets blocked for connection failure, _not_ auth failure. This can be simulated by just running `nc -q 0 -v <host> <port>` - the protocol handshake is server initiated and since we start talking to the db and then leave, it counts as a connection failure.
> 
> Default `max_connect_errors` is 100. I'm going to bump that value up a lot and then just wait for the db to get recreated over the weekend.

Sounds good. Hopefully we will see some data once performance_schema is enabled.

**Review by camscale — APPROVED** [review]
bot

**Pull request merged**